### PR TITLE
docs: Refactored HGETALL 

### LINKED ISSUE #830: Audit and make documentation for command `HGETALL` consistent
The documentation of the command `[HGETALL](https://dicedb.io/commands/hgetall/)` present in [the docs](https://github.com/DiceDB/dice/tree/master/docs/src/content/docs/commands) could have become stale. We need your help to ensure that the documentation is complete, correct, and consistent.  Completing this issue will give you much-needed exposure to how the command is implemented and the different parameters it accepts. Hence, this is a great way to build a good understanding of the project and its functionality. While still making a significant contribution. Here's what you need to do:

1. Go through the [documentation of the command](https://dicedb.io/commands/hgetall/) and run all the examples, making sure they all work as expected
2. If the same command is present in Redis, then the expected output of the DiceDB command is the same as the Redis output
3. If the command is not in Redis, then as per your judgment, [raise an issue with this template](https://github.com/DiceDB/dice/issues/new?assignees=&labels=&projects=&template=report_command_bug.md&title=Inconsistent+%60%7BCMD%7D%60%3A+%3CDescribe+the+error+in+one+concise+line%3E) or fix the documentation and raise a pull request.
4. The documentation should contain the following sections in the following order
 * short and concise introduction paragraph about the command covering what it does
 * Syntax
 * Parameters: List all the parameters the command accepts, and ignore the section if there are no parameters
 * Return values: List all possible return values and under what condition
 * Behaviour: Describe the behavior of the command, what it does, how it does it, some internal specifics if any
 * Errors: List all possible errors the command can throw and under what condition
 * Examples: List all possible examples of the command and the expected output. Assume CLI implementation

Note: The title of the sections should be the exact strings mentioned above, like "Examples", "Return values", etc.
Note: The description of the command in Frontmatter of the command.md file is the first paragraph (introduction) of the command.

Go through the [DiceDB/dice](https://github.com/dicedb/dice) repository and explore the command implementation
 * to understand all possible return values
 * to understand all possible error values and conditions
 * to understand different parameters the command accepts

Please use the documentation of the [SET](https://dicedb.io/commands/set/) command as the reference point and structure the
documentation of this command in a very similar way, including the following points
 * if you see any "Conclusion" section, remove it
 * the headers (h1, h2, h3) should be appropriately used
 * the CLI prompt should be `127.0.0.1:7379>` and not anything else
 * Use markdown tables for parameters and return values (ref SET command documentation)
 * wrap any command or paramter in backticks (`) to highlight them
 * the section header should be capitalize as used in the SET command documentation

If you find any inconsistencies, please fix the documentation and raise the PR.
The core idea of this exercise is to ensure that the documentation is consistent, correct, and complete.

Make sure you comment on the other issues you created (if any) as a comment on this issue and also any PR (if any) that you created.
Thank you for picking this up and contributing to the DiceDB. It means a ton.

### PR DESCRIPTION
**Pull request by sashpawar11**
This pr closes #830 

**Changes**
- Refactored sections
- Refactored error messages
- Added additional error messages and examples
- Refactored CLI syntaxes

### CODE REVIEW

**Comment by sashpawar11** [comment]
Hi @lucifercr07 @apoorvyadav1111 - please review.
Thankyou.

File: docs/src/content/docs/commands/HGETALL.md
Comment (AshwinKul28) [inline docs/src/content/docs/commands/HGETALL.md:92]: Instead of Example 4, can you please add title as Error Example and subtitle as `invalid number of arguments are passed`

**Review by AshwinKul28 — CHANGES_REQUESTED** [review]

**Comment by AshwinKul28** [comment]
Hi @sashpawar11 Thanks a lot for the changes, left one minor comment.

File: docs/src/content/docs/commands/HGETALL.md
Comment (apoorvyadav1111) [inline docs/src/content/docs/commands/HGETALL.md:27]: Hi, I am unable to find DiceDB's response for HGETALL with a non existent key saying "set". I am running latest master on my local. Please confirm.

File: docs/src/content/docs/commands/HGETALL.md
Comment (apoorvyadav1111) [inline docs/src/content/docs/commands/HGETALL.md:35]: same as above.

File: docs/src/content/docs/commands/HGETALL.md
Comment (apoorvyadav1111) [inline docs/src/content/docs/commands/HGETALL.md:43]: same as above

File: docs/src/content/docs/commands/HGETALL.md
Comment (apoorvyadav1111) [inline docs/src/content/docs/commands/HGETALL.md:75]: This looks inconsistent to what I am getting on latest DiceDB and redis locally. Please confirm.

**Review by apoorvyadav1111 — CHANGES_REQUESTED** [review]
Hi, Thanks for the changes, have left some queries, which are related to each other.

**Comment by sashpawar11** [comment]
Hey @sashpawar11 - a PR is already raised and linked to this ticket.

There are some changes suggested on the PR, will try to resolve those and get it merged asap.

Thankyou.

**Comment by lucifercr07** [comment]
@sashpawar11 can you please address review comments from @apoorvyadav1111

File: docs/src/content/docs/commands/HGETALL.md
Comment (sashpawar11) [reply docs/src/content/docs/commands/HGETALL.md:75]: Hey @apoorvyadav1111 - addressing all the comments mentioned above, I haven't taken the latest pull and the current state of my fork is of that when this PR was made, and here's the same scenario I executed today:

image

Comment (sashpawar11) [reply docs/src/content/docs/commands/HGETALL.md:75]: Took the latest pull, getting the same response as above

Comment (apoorvyadav1111) [reply docs/src/content/docs/commands/HGETALL.md:75]: I think the difference comes from using redis-cli or dice-cli. I would say keep it as empty array for now to keep it consistent across other docs.

Comment (sashpawar11) [reply docs/src/content/docs/commands/HGETALL.md:75]: Cool @apoorvyadav1111 - keeping it empty array, lets keep it consistent as you mentioned. Will push the recommit soon.
Thanks

File: docs/src/content/docs/commands/HGETALL.md
Comment (sashpawar11) [reply docs/src/content/docs/commands/HGETALL.md:27]: Resolved

File: docs/src/content/docs/commands/HGETALL.md
Comment (sashpawar11) [reply docs/src/content/docs/commands/HGETALL.md:35]: Resolved

**Comment by sashpawar11** [comment]
Ughh it didn't fast-forward somehow, please advise if this is fine or should i drop the pr and make a new one 
@apoorvyadav1111 @AshwinKul28

**Review by apoorvyadav1111 — APPROVED** [review]
Thanks for the changes, approving the PR. Kindly wait for all reviewers' approval for merge.
Thanks

**Review by lucifercr07 — APPROVED** [review]

**Pull request merged**
